scripts/cleanup-test-processes.sh

```diff
@@ -1,35 +1,37 @@
 #!/usr/bin/env bash
 # This script can be handy to kill test processes launched manually
-# for debugging or still hanging around for other reasons.
+# for debugging or that are still hanging around for other reasons. If
+# you have monero-wallet-rpc or swapd processes owned by the current
+# user that you don't want killed, don't use this script!
 
 pkill_cmd=(pkill --echo --uid "${UID}" --full)
 if [[ "$(uname)" == 'Darwin' ]]; then
 	pkill_cmd=(pkill -l -U "${UID}" -f)
 fi
 
-echo "Stopping any monerod regest instances"
-"${pkill_cmd[@]}" '/monerod .* --regtest '
-
-echo "Stopping any ganache instances"
-"${pkill_cmd[@]}" '/ganache '
-
-echo "Stopping any relayer instances"
-"${pkill_cmd[@]}" '/relayer '
-
-# If you have monero-wallet-rpc or swapd processes owned by the current user
-# that you don't want to kill, don't use this script!
 echo "Stopping any swapd instances"
 if "${pkill_cmd[@]}" '/swapd '; then
-	# If swapd instances were killed, give the monero-wallet-rpc instances
+	# If swapd instances were killed, give dependant monero-wallet-rpc instances
 	# some time to shutdown.
 	sleep 4 # Give time for wallet to shutdown
 fi
 
+echo "Stopping any relayer instances"
+"${pkill_cmd[@]}" '/relayer '
+
 # Take note of monero-wallet-rpc instances being killed here if the instances
-# were started by swapd
+# were started by swapd, as killing swapd should have already shut them down.
 echo "Stopping any monero-wallet-rpc instances"
 "${pkill_cmd[@]}" '/monero-wallet-rpc '
 
+echo "Stopping any monerod regest instances"
+if "${pkill_cmd[@]}" '/monerod .* --regtest '; then
+	sleep 2 # we don't want to exit the script while it is still running
```

Comment:
I put this sleep here and removed it from `setup-env.sh`. The other changes were just reordering, so that we kill dependent processes first before the processes they depend on.